python/fastquant/data/forex/forextester.py

```diff
@@ -0,0 +1,483 @@
+import os
+from pathlib import Path
+import zipfile
+import logging
+import pickle
+import multiprocessing
+
+import requests
+import pandas as pd
+from pandas import DataFrame
+
+from fastquant.config import DATA_PATH
+from fastquant.data.stocks.pse import datestring_to_datetime
+
+handler_format = logging.Formatter('%(asctime)s-%(name)s-%(levelname)s-%(message)s',datefmt='%Y-%m-%d %H:%M:%S')
+logger = logging.getLogger('forex_logger')
+logger_handler = logging.StreamHandler()
+logger_handler.setFormatter(handler_format)
+logger.setLevel(logging.INFO)
+logger.addHandler(logger_handler)
+
+
+def get_forextester_data(symbol, start_date=None, end_date=None,  time_frame='D1'):
+    allowed_forex_symbol_list = [
+        'AUDJPY',
+        'AUDUSD',
+        'CHFJPY',
+        'EURCAD',
+        'EURCHF',
+        'EURGBP',
+        'EURJPY',
+        'EURUSD',
+        'GBPCHF',
+        'GBPJPY',
+        'GBPUSD',
+        'NZDJPY',
+        'NZDUSD',
+        'USDCAD',
+        'USDJPY',
+        'USDCHF',
+        'XAGUSD',
+        'XAUUSD'
+    ]
+    if symbol not in allowed_forex_symbol_list:
+        raise ValueError('your symbol is not supported by forextester')
+
+    if start_date is not None:
+        start_date = datestring_to_datetime(start_date)
+        end_date = datestring_to_datetime(end_date)
+
+    logger.info(f'start downloading forex {symbol} data zip file...')
+    file_size = 0
+    res = requests.get(f'http://www.forextester.com/templates/data/files/{symbol}.zip', stream=True)
```

Comment:
For the first time to use, it will download data from online and make local cache, after that set get_forex_data's parameter read_from_local=False so that it will read data from local.

Reply:
I agree with enzoampil that we should avoid the caching all forex data. We should cache only what is requested. Is there an easy implementation you can try?

Reply:
Makeing one symbol and one period cache in single process needs 15 minutes and 2GB RAM, is the cost acceptable?
A minimum of 2GB of memory is required to load 1 min chart of forex data from 2000 to 2018.

Reply:
Can you explain why such large memory is needed?